### PR TITLE
Fix/refactor

### PR DESCRIPTION
**Pull request by mspector**
adds proper error handling to OpenReviewPy

### CODE REVIEW

File: openreview/openreview.py [hunk 1]
```diff
@@ -17,7 +18,7 @@ def __init__(self, baseurl=None, process_dir='../process/', webfield_dir='../web
             try:
                 self.baseurl = os.environ['OPENREVIEW_BASEURL']
             except KeyError:
-                self.baseurl = 'http://localhost:3000'
+                self.baseurl = 'http://openreview.net'
```
Comment (mspector) [inline openreview/openreview.py]: this is a significant change, from localhost --> openreview

I'm interested to hear your opinion on this change. It makes sense because most users of openreviewpy will only ever interact with the openreview.net instance. But it means that, when we are testing things, we'll have to make sure that we're using localhost

File: openreview/openreview.py [hunk 2]
```diff
@@ -298,92 +249,65 @@ def get_notes(self, id=None, forum=None, invitation=None, replyto=None, tauthor=
             params['trash']=True
 
         response = requests.get(self.notes_url, params=params, headers=self.headers)
-        
+        response = self.__handle_response(response)
+
+        for n in response.json()['notes']:
+            note = Note(n['id'],
```
Comment (melisabok) [inline openreview/openreview.py]: Could you move this code to a class method from_json?
Something like this:

```
    @classmethod
    def from_json(json_string):
          
```

File: openreview/openreview.py [hunk 3]
```diff
@@ -17,7 +18,7 @@ def __init__(self, baseurl=None, process_dir='../process/', webfield_dir='../web
             try:
                 self.baseurl = os.environ['OPENREVIEW_BASEURL']
             except KeyError:
-                self.baseurl = 'http://localhost:3000'
```
Comment (melisabok) [inline openreview/openreview.py]: I would keep the localhost by default, and force the user to specify the baseUrl.

File: openreview/openreview.py (hunk 2 above)
Comment (mspector) [inline openreview/openreview.py]: yes, do you want it in this PR? or should I make another PR for this change?

Comment (melisabok) [inline openreview/openreview.py]: can you do it in this PR?

File: openreview/openreview.py [hunk 4]
```diff
@@ -240,41 +147,14 @@ def get_invitations(self, id=None, invitee=None, replytoNote=None, replyForum=No
...
-                    return error
-            except KeyError:
-                for error in response.json()['error']:
-                    return error
+        response = self.__handle_response(response)
+
+        for i in response.json()['invitations']:
+            invitation = Invitation.from_json(i)
+            invitations.append(invitation)
+            invitations.sort(key=lambda x: x.id)
```
Comment (melisabok) [inline openreview/openreview.py]: The sort should be outside the loop

File: openreview/openreview.py [hunk 5]
```diff
@@ -192,35 +118,16 @@ def get_groups(self, prefix=None, regex=None, member=None, host=None, signatory=
...
-                
-        except requests.exceptions.HTTPError as e:
-            try:
-                for error in response.json()['errors']:
-                    return error
-            except KeyError:
-                for error in response.json()['error']:
-                    return error
+        for g in response.json()['groups']:
+            g = self.get_group(g['id']).to_json()
```
Comment (melisabok) [inline openreview/openreview.py]: do we still need to call get_group for each group of the response?

File: openreview/openreview.py [hunk 6]
```diff
@@ -298,92 +178,51 @@ def get_notes(self, id=None, forum=None, invitation=None, replyto=None, tauthor=
             params['trash']=True
 
         response = requests.get(self.notes_url, params=params, headers=self.headers)
-        
+        response = self.__handle_response(response)
+
+        for n in response.json()['notes']:
+            note = Note.from_json(n)
+            notes.append(note)
+        notes.sort(key=lambda x: x.forum)
```
Comment (melisabok) [inline openreview/openreview.py]: Sort outside the loop, btw the notes service returns the notes sorted by date. I don't why you want to sort by forum

Comment (mspector) [inline openreview/openreview.py]: hmm I'm not sure either. I'll take it out

File: openreview/openreview.py (hunk 5 above)
Comment (mspector) [inline openreview/openreview.py]: i just tested it, and it looks like we don't

**Comment by melisabok** [comment]
Excellent!!!! Library is getting better and better 💯
